models/sessions/scratch/snowplow_mobile_sessions_aggs.sql

```diff
@@ -11,39 +11,43 @@
 }}
 
 with events as (
- select
-      es.session_id,
-      es.event_id,
-      es.event_name,
-      es.derived_tstamp,
-      es.build,
-      es.version,
-      es.event_index_in_session,
-      MAX(es.event_index_in_session) over (partition by es.session_id) as events_in_session
+  select
+    es.session_id,
+    es.event_id,
+    es.event_name,
+    es.derived_tstamp,
+    es.build,
+    es.version,
+    es.event_index_in_session,
+    MAX(es.event_index_in_session) over (partition by es.session_id) as events_in_session
 
-    from {{ ref('snowplow_mobile_base_events_this_run') }}   es
+  from {{ ref('snowplow_mobile_base_events_this_run') }}   es
 )
 
 , session_aggs AS (
-    select
-      e.session_id,
-      --last dimensions
-      MAX(case when e.event_index_in_session = e.events_in_session then e.build end) as last_build,
-      MAX(case when e.event_index_in_session = e.events_in_session then e.version end) as last_version,
-      MAX(case when e.event_index_in_session = e.events_in_session then e.event_name end) as last_event_name,
+  select
+    e.session_id,
+    --last dimensions
+    MAX(case when e.event_index_in_session = e.events_in_session then e.build end) as last_build,
+    MAX(case when e.event_index_in_session = e.events_in_session then e.version end) as last_version,
+    MAX(case when e.event_index_in_session = e.events_in_session then e.event_name end) as last_event_name,
+    {% if target.type == 'postgres' %}
+      MAX(case when e.event_index_in_session = e.events_in_session then e.event_id::varchar end) as session_last_event_id,
```

Comment:
`event_id` and `session_id` are both `uuid`s and for some reason Postgres can't deal with that. So in the join it can't implictly cast a `uuid` to a `varchar` to compare to another `varchar`, and it also doesn't know a `max` function for a `uuid`.
For `app_errors`, it sees that column as `text` unless I explicitly cast it. I've been able to cast `app_errors` upstream in the `snowplow_mobile_sessions_this_run` model successfully and have it trickle down, I've done the same for `session_id` in `snowplow_mobile_base_session_context` and `snowplow_mobile_base_sessions_lifecycle_manifest` but for `event_id` I can either do it in `snowplow_mobile_sessions_aggs` or I'd have to unstar in `snowplow_mobile_base_events_this_run` and do it there. I've done the former for now but let me know if you prefer the latter